.github/workflows/test-build-deploy.yml

```diff
@@ -278,17 +278,21 @@ jobs:
       - name: Build Mimir with race-detector
         run: |
           export IMAGE_TAG_RACE=$(make image-tag-race)
-          export MIMIR_IMAGE="grafana/mimir:$IMAGE_TAG_RACE"
+          export MIMIR_IMAGE="grafana/mimir:distroless-$IMAGE_TAG_RACE"
+          export MIMIR_ALPINE_IMAGE="grafana/mimir:$IMAGE_TAG_RACE"
           make BUILD_IN_CONTAINER=false cmd/mimir/.uptodate_race
           docker save $MIMIR_IMAGE -o ./mimir_race_image
+          docker save $MIMIR_ALPINE_IMAGE -o ./mimir_race_image_alpine
       - name: Upload archive with race-enabled Mimir
         uses: actions/upload-artifact@v4
         with:
           name: Race-enabled Mimir
-          path: ./mimir_race_image
+          path: |
+            ./mimir_race_image
+            ./mimir_race_image_alpine
 
   integration:
```

Comment:
The more I look at this, the more I think the integration job needs to be ran separately for distroless vs alpine. That is, at a job level, not just at a step level.

This could be done with the matrix below:
```
      matrix:
        image: [distroless, alpine]
        test_group_id:    [0, 1, 2, 3, 4, 5]
        test_group_total: [6]
```

Unfortunately a job's `if` is [evaluated before the matrix is evaluated](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif). So we would then need to do an if on each of the steps within the job:

`if: env.is_deploy || matrix.image == 'distroless'`

Reply:
I doubt we should do matrix on image, since to run integration test, we need to export Mimir_image, to run it on parallel is confusing, the 2 jobs would fight to set the environement variable to distroless and alpine and distroless. Or I missed something about metrics it is a local environment variable?

Reply:
PR updated with metrics